backend/de.metas.deliveryplanning.base/src/main/java/de/metas/deliveryplanning/process/M_Delivery_Planning_GenerateShortageOverage.java

```diff
@@ -0,0 +1,264 @@
+		{
+			return ProcessPreconditionsResolution.rejectWithInternalReason("Delivery planning is not of Type Incoming");
+		}
+
+		final DeliveryPlanningReceiptInfo receiptInfo = optionalDeliveryPlanningReceipt.get();
+		if (!receiptInfo.isReceived())
+		{
+			return ProcessPreconditionsResolution.rejectWithInternalReason("Not received");
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Nullable
+	@Override
+	public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
+	{
+		if(PARAM_QTY.contentEquals(parameter.getColumnName()))
+		{
+			return PARAM_DEFAULT_VALUE_ZERO;
+		}
+		return IProcessDefaultParametersProvider.DEFAULT_VALUE_NOTAVAILABLE;
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final DeliveryPlanningId deliveryPlanningId = DeliveryPlanningId.ofRepoId(getRecord_ID());
+
+		final DocTypeQueryBuilder docTypeQuery = DocTypeQuery.builder()
+			.docBaseType(MaterialPhysicalInventory)
+			.adClientId(getClientID().getRepoId())
+			.adOrgId(OrgId.ANY.getRepoId());
+
+		final AdMessageKey adMessageKey;
+		if (p_Qty.signum() > 0)
+		{
+			adMessageKey = MSG_Event_OverageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryOverageDocument);
+		}
+		else if (p_Qty.signum() < 0)
+		{
+			adMessageKey = MSG_Event_ShortageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryShortageDocument);
+		}
+		else
+		{
+			throw new AdempiereException(MSG_ERROR_QUANTITY_ZERO);
+		}
+
+		final I_M_Inventory inventory = generateInventoryDocument(docTypeDAO.getDocTypeId(docTypeQuery.build()), deliveryPlanningId);
+		createUserNotification(inventory, adMessageKey);
+
+		return MSG_OK;
+	}
+
+	private I_M_Inventory generateInventoryDocument(@NonNull final DocTypeId doctypeId, @NonNull final DeliveryPlanningId deliveryPlanningId)
+	{
+		final DeliveryPlanningReceiptInfo receiptInfo = deliveryPlanningService.getReceiptInfo(deliveryPlanningId);
+		final InOutId inOutId = receiptInfo.getReceiptId();
+		Check.assumeNotNull(inOutId, "InOutId shall be set, because of isReceived() check in preconditions");
+
+		final I_M_InOut inOutRecord = load(inOutId, I_M_InOut.class);
+		final List<I_M_InOutLine> inOutLines = inOutDAO.retrieveLines(inOutRecord);
+		if(inOutLines.size() == 0)
+		{
+			throw new AdempiereException("No receipt lines found");
+		}
+		if(inOutLines.size() > 1)
+		{
+			throw new AdempiereException("More than 1 receipt line exists");
+		}
+
+		final I_M_Inventory inventoryRecord = newInstance(I_M_Inventory.class);
+		inventoryRecord.setAD_Org_ID(receiptInfo.getOrgId().getRepoId());
+		inventoryRecord.setC_DocType_ID(doctypeId.getRepoId());
+		inventoryRecord.setM_Warehouse_ID(inOutRecord.getM_Warehouse_ID());
+		inventoryRecord.setMovementDate(inOutRecord.getMovementDate());
+
+		saveRecord(inventoryRecord);
+		logger.trace("Insert inventory - {}", inventoryRecord);
+
+		createInventoryLine(InventoryId.ofRepoId(inventoryRecord.getM_Inventory_ID()), inOutLines.get(0));
```

Comment:
1st of all: use Collection.singleElement()
2nd of all: you don't need to handle I_M_InOutLine - as i said before, please try to use them as little as possible outside of repos. In thista case you just need the `de.metas.inout.InOutLineId` in order to create your `TableRecordReference`